Update metric check for resnet18 dynamo unit test

wonjoolee95 opened the pull request:
Makes test compatible with https://github.com/pytorch/pytorch/pull/117434.

JackCaoG commented on test/dynamo/test_dynamo.py:
```diff
@@ -502,7 +502,9 @@ def test_resnet18(self):
-    self.assertEqual(met.metric_data('CompileTime')[0], 5)
+    # Graph 6, 7: PyTorch has updated the number of captured by resnet with
+    # https://github.com/pytorch/pytorch/pull/117434. 
+    self.assertEqual(met.metric_data('CompileTime')[0], 7)
```
do we know why we compile 2 more graphs?

mlazos replied:
Yes because previously SGD was disabled, now it's enabled :)

JackCaoG replied:
oh ok, that make sense!

mlazos commented on test/dynamo/test_dynamo.py:
```diff
@@ -502,7 +502,9 @@ def test_resnet18(self):
+    # (https://github.com/pytorch/pytorch/pull/117434)
+    self.assertEqual(met.metric_data('CompileTime')[0], 7)
     # We execute 4 graphs per step when optimizer is enabled.
     self.assertEqual(met.metric_data('ExecuteTime')[0], sample_count * 4)
```
Will this number also need to be changed?

wonjoolee95 replied:
Yep, it should -- I'm re-running the whole test suite will correct all the values.

wonjoolee95 commented:
@mlazos, this should be ready now. The process for merging these PRs should be:
1. Update the PyTorch PR (https://github.com/pytorch/pytorch/pull/117434) XLA pin to the commit pin of this PR -- `7d1e8e7597519eaae130a909f0b4d391ab1c5c32`.
2. Once PyTorch PR CI is green, merge it with the pin.
3. Then I'll rebase this PyTorch/XLA PR with PyTorch head and merge.

mlazos commented:
> @mlazos, this should be ready now. The process for merging these PRs should be:
> 
> 1. Update the PyTorch PR ([Re-enable SGD pytorch#117434](https://github.com/pytorch/pytorch/pull/117434)) XLA pin to the commit pin of this PR -- `7d1e8e7597519eaae130a909f0b4d391ab1c5c32`.
> 2. Once PyTorch PR CI is green, merge it with the pin.
> 3. Then I'll rebase this PyTorch/XLA PR with PyTorch head and merge.

Hmm not sure if this needs to be rebased again, but I think another PR went in recently that also updated the XLA pin. https://github.com/pytorch/pytorch/commit/208e64a9ba2abca63ea77b3b2dc5a2f19e4c2ac4

JackCaoG commented:
lol yea, we had similar issue on the other pr and we merged that one first. Can you rebase?

JackCaoG commented:
Oh ok let me be more clear, I think we need to first
1. rebase this pr 
2. rebase pytorch pr to point to the new commit

This way your pytorch pr will point to a commit that can pass all CIs

mlazos commented:
@wonjoolee95 can you rebase?

wonjoolee95 commented:
> @wonjoolee95 can you rebase?

Yep, just rebased with master.

mlazos commented:
@wonjoolee95 Can we rebase again lol, I fixed the issue that reverted the PR, and need to udpate the xla pin again

wonjoolee95 commented:
> @wonjoolee95 Can we rebase again lol, I fixed the issue that reverted the PR, and need to udpate the xla pin again

Sure thing! Just rebased -- also not that with each rebase, the commit pin will change. Now latest commit pin to pin is `2990cb38c17e06d0dbe25437674ca40130d76a8f`.

mlazos commented:
https://github.com/pytorch/pytorch/pull/117434 landed finally, should be good to update the pin on this side

wonjoolee95 commented:
Merging this now to unblock head. Previously, CI was all green and locally the dynamo unit tests are passing.
